cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -331,13 +389,18 @@ where
 		Some(CollationResult { collation: Collation { head_data, number }, result_sender: None })
 	}
 
-	async fn produce_bundle(self, slot_info: ExecutorSlotInfo) -> Option<BundleResult> {
+	async fn produce_bundle(
+		self,
+		primary_hash: PHash,
+		slot_info: ExecutorSlotInfo,
+	) -> Option<BundleResult> {
 		println!("TODO: solve some puzzle based on `slot_info` to be allowed to produce a bundle");
 
-		// TODO: ready at the best number of primary block?
-		let block_number = self.client.info().best_number;
+		let parent_hash = self.client.info().best_hash;
+		let parent_number =
+			self.client.expect_block_number_from_id(&BlockId::Hash(parent_hash)).ok()?;
```

Comment:
Wouldn't this work too?:
```suggestion
		let info = self.client.info();
		let parent_hash = info.best_hash;
		let parent_number = info.best_number;
```

Reply:
I thought about that too, but not 100% sure, so I just used this safe version. Since you raised this question, I check it again and think it should work as it's already used in https://github.com/paritytech/substrate/blob/cd3d2b1b65929bafa699953094bcd8b9fe7d3dfc/client/service/src/client/client.rs#L1314-L1315